### DISCUSSION TITLE
Just applause

### DISCUSSION THREAD

**Discussion by kubycsolutions** [cardonabits/haxo-hw#11] [General]
As shown by the demo, you do seem to have succeeded nicely.

I suspect it isn't responsive enough to do the sing-for-overtines effect, but can you get a growl out of it?

Possible feature: MIDI output, as alternative to or in parallel with the built-in sound generator. Breath should map to some combination of velocity and after touch, I think...

This is reminding me of Casio's toy electronic sax, which I think actually used something closer to recorder fingerings when in friendly-to-musicians mode. It also had a friendly-to-programmers mode where the right hand played a gray-code scale on three fingers and the left hand had shifts for sharp, flat, and two-bit gray-code octave shifts, giving a four octave range plus a half note at top and bottom. I found the latter surprisingly playable. It had midi out but no dynamics, not even silence when not blown, which made that rather useless and motivates my wish for passing dynamics thru the midi when and if added.

Now I need to find time to build my own instrument design. Thanks for the inspiration!

**Comment by jcard0na** [comment]
Thank you very much for the encouragement, Joe! I'll look up that Casio toy sax, seems like a fun toy.

Regarding MIDI output, this would be relatively easy to implement.  Internally the software does convert fingerings to [midi notes](https://github.com/jcard0na/haxo-rs/blob/main/src/midinotes.rs), and air pressure [to volume](https://github.com/jcard0na/haxo-rs/blob/main/src/main.rs#L87), which is then passed to the software MIDI synth fluidsynth.  These events could be sent out of a MIDI connector, if someone wanted to modify the PCB to add one.  😉
